node/node.go

```diff
@@ -155,21 +155,36 @@ func NewNode(ctx context.Context,
 		DiscardABCIResponses: config.Storage.DiscardABCIResponses,
 	})
 
-	genDoc, err := genesisDocProvider()
+	// Get the genesisDoc hash from db if exists
+	genDocHash, err := stateDB.Get(genesisDocHashKey)
```

Comment:
Note that, if the key did not exist there will be a `errKeyEmpty` or something like that return. Worth double checking whether that breaks the logic.

Reply:
It's handled the same as it was prior to this PR: https://github.com/cometbft/cometbft/pull/1293/files#diff-bbaa90ad1c80edb1f0084525de9b5e22da605e22d5e0b374d04c1e6a363d6f9eL543
If this is wrong, let's fix it here.
I actually assumed db `Get` method gives us an empty array and no error if the key is missing. Need to learn our internal APIs.

Reply:
You are right, I checked , the db wraps the error in this case and does not return it but rather just retuns a `nil` value so this is all good.